tools/tpm2_ptool.py

```diff
@@ -1063,175 +1291,90 @@ def generate_options(self, group_parser):
-                    sealpriv = sealobject['userpriv']
-                    salt = sealobject['userauthsalt']
-                    iters = sealobject['userauthiters']
+        with TemporaryDirectory() as d:
+            tpm2 = Tpm2(d, path)
+            tertiarypriv, tertiarypub, tertiarypubdata = tpm2.importkey(sobjctx, sobjauth, objauth, privkey=privkey, alg=alg)
 
-                salt = binascii.unhexlify(salt)
-                sealauth = hash_pass(pin.encode(), iters, salt)['hash']
+            return (tertiarypriv, tertiarypub, tertiarypubdata)
 
-                # Load the so sealobject using the PARENTS AUTH (primaryobject)
-                sealctx = tpm2.load(pobj['handle'], pinpobjauth, sealpriv, sealpub)
+    def __call__(self, args):
+        super(self.__class__, self).__call__(args)
+        keylabel= args['key_label']
+        print('Imported key as label: "{keylabel}"'.format(keylabel=keylabel))
 
-                # Now that the sealobject is loaded, we need to unseal the wrapping key
-                # object auth or the key when the TPM doesn't support encryptdecrypt
 
-                wrappingauth = tpm2.unseal(sealctx, sealauth)
+@commandlet("addkey")
+class AddKeyCommand(NewKeyCommandBase):
+    '''
+    Adds a key to a token within a tpm2-pkcs11 store.
+    '''
 
-                sym_support = str2bool(token_config['sym-support'])
-                if sym_support:
-                    # Now that we have the unsealed wrappingauth value
-                    # Load the wrapping object
-                    wrappingobj = db.getwrapping(token['id'])
-                    wrappingctx = tpm2.load(pobj['handle'], pinpobjauth, wrappingobj['priv'], wrappingobj['pub'])
+    # adhere to an interface
+    # pylint: disable=no-self-use
+    def generate_options(self, group_parser):
+        group_parser.add_argument(
+            '--label',
+            help='The tokens label to add a key too.\n',
+            required=True)
+        group_parser.add_argument(
+            '--algorithm',
+            help='The type of the key.\n',
+            choices=[ 'rsa1024', 'rsa2048', 'aes128', 'aes256' ],
+            required=True)
+        group_parser.add_argument(
+            '--key-label',
+            help='The key label to identify the key. Defaults to an integer value')
+        group_parser.add_argument(
+            '--id',
+            help='The key id. Defaults to a random 8 bytes of hex.\n',
+            default=binascii.hexlify(os.urandom(8)))
+        pinopts = group_parser.add_mutually_exclusive_group(required=True)
+        pinopts.add_argument(
+            '--sopin',
+            help='The Administrator pin,'),
+        pinopts.add_argument(
+            '--userpin',
+            help='The User pin.\n'),
 
-                # Now get the secondary object from db
-                sobj = db.getsecondary(token['id'])
+    # Creates a new key
+    def new_key_create(self, args, sobjctx, sobjauth, objauth, tpm2):
```

Comment:
this is weird that theirs arguments coming from a dictionary as well as named local arguments. Unpack args throughout.

Reply:
Can you elaborate please? looks like something python specific?
I'm guessing I need to use store all the argument in a list and then use it while calling the method?